packages/backend/src/services/RenameService/RenameService.ts

```diff
@@ -0,0 +1,695 @@
+
+    private readonly schedulerClient: SchedulerClient;
+
+    private readonly schedulerModel: SchedulerModel;
+
+    constructor(args: RenameServiceArguments) {
+        super();
+        this.lightdashConfig = args.lightdashConfig;
+        this.analytics = args.analytics;
+        this.savedChartModel = args.savedChartModel;
+        this.projectModel = args.projectModel;
+        this.dashboardModel = args.dashboardModel;
+        this.schedulerClient = args.schedulerClient;
+        this.schedulerModel = args.schedulerModel;
+    }
+
+    public async getFieldsForChart({
+        user,
+        projectUuid,
+        chartUuid,
+    }: {
+        user: SessionUser;
+        projectUuid: string;
+        chartUuid: string;
+    }) {
+        const chart = await this.savedChartModel.get(chartUuid);
+        const explore = await this.projectModel.getExploreFromCache(
+            projectUuid,
+            chart.tableName,
+        );
+        if (isExploreError(explore)) {
+            throw new NotFoundError(
+                `Valid explore ${chart.tableName} not found`,
+            );
+        }
+        // TODO also add table calculations, custom metrics and dimensions
+        // TODO also add joins
+        const fields = Object.keys(
+            explore.tables[explore.baseTable].dimensions,
+        ).concat(Object.keys(explore.tables[explore.baseTable].metrics));
+        return {
+            [explore.baseTable]: fields.map((f) =>
+                getItemId({ name: f, table: explore.baseTable }),
+            ), // add explore prefix
+            tableCalculations: chart.metricQuery.tableCalculations?.map(
+                (tc) => tc.name,
+            ),
+            customMetrics: chart.metricQuery.additionalMetrics?.map(
+                (tc) => tc.name,
+            ),
+            customDimensions: chart.metricQuery.customDimensions?.map(
+                (tc) => tc.name,
+            ),
+        };
+    }
+
+    /**
+     * Triggered from the UI on validation settings
+     */
+    async renameChart({
+        user,
+        projectUuid,
+        chartUuid,
+        from,
+        to,
+        type,
+        fixAll,
+        context,
+    }: ApiRenameChartBody & {
+        user: SessionUser;
+        projectUuid: string;
+        context: RequestMethod;
+    }) {
+        if (from === to) {
+            throw new ParameterError(
+                'Old and new names are the same, nothing to rename',
+            );
+        }
+        const { organizationUuid } = await this.projectModel.getSummary(
+            projectUuid,
+        );
+        if (
+            user.ability.cannot(
+                'update',
+                subject('Project', {
+                    organizationUuid,
+                    projectUuid,
+                }),
+            )
+        ) {
+            throw new ForbiddenError();
+        }
+
+        const chart = await this.savedChartModel.get(chartUuid);
```

Comment:
does update project permission assume admin? in that case no permission check is required here

Reply:
update project is `developers`

We agreed this is ok for developers to use, even if they don't have access to the chat because of the space (more details in slack)